core/repository/referredUsers/tree/FactorNode.ts

```diff
@@ -1,40 +1,36 @@
 import config from 'shared/config/index';
 import { Node } from 'core/util/tree';
-import { FactorType, Factors } from 'core/repository/referredUsers/interfaces';
+import { FactorType, ReferredUserFactor, FactorAction } from 'core/repository/referredUsers/interfaces';
 
 export default class FactorNode<T> extends Node<T> {
 
-    levelSummary: Array<Factors>;
+    private readonly levelSummary: Array<ReferredUserFactor>;
 
     constructor(data: T) {
         super(data);
 
-        this.levelSummary = new Array(config.CONSTANTS.REFERRAL.MAX_COUNT);
+        this.levelSummary = new Array(config.CONSTANTS.REFERRAL.MAX_COUNT + 1);
 
-        for (let i = 0; i < config.CONSTANTS.REFERRAL.MAX_COUNT; i++) {
-            this.levelSummary[i] = this.getEmptyFactor();
+        for (let i = 1; i <= config.CONSTANTS.REFERRAL.MAX_COUNT; i++) {
+            this.levelSummary[i] = ReferredUserFactor.createEmpty();
         }
     }
 
-    getEmptyFactor(): Factors {
-        return new Map([
-            [FactorType.COUNT, 0],
-            [FactorType.REWARD, 0],
-            [FactorType.STAKE_AMOUNT, 0]
-        ]);
+    private idLevelValid(level: number): boolean {
+        return level >= 1 && level <= config.CONSTANTS.REFERRAL.MAX_COUNT;
```

Comment:
I think in memory we should store `level` like a `level - 1`

Reply:
I wanted the level to match the height in the tree